frontend/controller/actions/chatroom.js

```diff
@@ -21,15 +21,16 @@ sbp('okTurtles.events/on', MESSAGE_RECEIVE_RAW, ({
   // If newMessage is undefined, it means that an existing message is being edited
   newMessage
 }) => {
-  const getters = sbp('state/vuex/getters')
-  const mentions = makeMentionFromUserID(getters.ourIdentityContractId)
+  const state = sbp('chelonia/contract/state', contractID)
+  const rootState = sbp('chelonia/rootState')
+  const mentions = makeMentionFromUserID(rootState.loggedIn?.identityContractID)
   const msgData = newMessage || data
   const isMentionedMe = (!!newMessage || data.type === MESSAGE_TYPES.TEXT) && msgData.text &&
   (msgData.text.includes(mentions.me) || msgData.text.includes(mentions.all))
 
   if (!newMessage) {
-    const isAlreadyAdded = !!getters
-      .chatRoomUnreadMessages(contractID).find(m => m.messageHash === data.hash)
+    // TODO:  rootState.unreadMessages for SW
+    const isAlreadyAdded = rootState.chatroom?.unreadMessages?.[contractID]?.unreadMessages.find(m => m.messageHash === data.hash)
```

Comment:
Is it possible to keep the getter?

Reply:
The reason that the getter isn't being used here is that although I added a wrapper for root getters, these are Vuex modules, which work a bit differently due to state partitioning. At the time, the separate state keys being used by the module weren't there. Now they are, and it would be feasible to similarly have this work.

While I think that _in this instance_ it'd be possible to use a getter, many getters are written to rely on `curentXXXId`, which is problematic in the SW.